src/Illuminate/Foundation/Console/RouteCacheCommand.php

```diff
@@ -58,12 +61,28 @@ public function handle()
             return $this->error("Your application doesn't have any routes.");
         }
 
-        foreach ($routes as $route) {
+        $symfonyRoutes = new SymfonyRouteCollection();
+
+        foreach ($routes as &$route) {
             $route->prepareForSerialization();
+
+            // If the route doesn't have a name, we'll generate one for it
+            // and re-add the route to the collection. This way we can
+            // add the route to the Symfony route collection.
+            if (! $name = $route->getName()) {
+                $route->name($name = Str::random());
+
+                $routes->add($route);
```

Comment:
For the new system I needed every route to have a name so if a route doesn't has a name, I generate one for it. I don't know if this has complications for any existing app. Does anyone have any thoughts if this could affect some apps?